src/world/Action/EffectResult.cpp

```diff
@@ -47,14 +48,36 @@ void EffectResult::damage( uint32_t amount, Common::ActionHitSeverityType severi
   m_type = Common::ActionEffectType::Damage;
 }
 
-void EffectResult::heal( uint32_t amount, Sapphire::Common::ActionHitSeverityType severity )
+void EffectResult::heal( uint32_t amount, Sapphire::Common::ActionHitSeverityType severity, bool isSelfHeal )
 {
   m_severity = severity;
   m_value = amount;
+  m_flag = isSelfHeal ? 0x80 : 0; // flag == 0x80 displays healing text at source actor
 
   m_type = Common::ActionEffectType::Heal;
 }
 
+void EffectResult::restoreMP( uint32_t amount )
+{
+  m_value = amount;
+  m_flag = 0x80;
+
+  m_type = Common::ActionEffectType::MpGain;
+}
+
+void EffectResult::startCombo( uint16_t actionId )
+{
+  m_value = actionId;
+  m_flag = 0x80;
+
+  m_type = Common::ActionEffectType::StartActionCombo;
+}
+
+void EffectResult::comboVisualEffect()
```

Comment:
is startCombo and comboVisualEffect mutually exclusive or something? maybe these names are a bit vague or I'm misunderstanding, but shouldn't there always be a visual effect? iirc when sending combos with effect you always had to send the successful combo action id

Reply:
StartActionCombo makes client flashes the next combo skill icon and ComboVisualEffect makes client uses a different skill effect (animation?) than using it uncomboed.

Reply:
ah okay, so there'd be instances where you'd only have the visual effect, sounds good